### PR TITLE
tracing: add options to NewFromEnv

### PR DESCRIPTION
**Pull request by hairyhenderson**
I've found myself wanting to set some options (tracer tags, etc), and figured this should work fine without breaking existing code.

Signed-off-by: Dave Henderson <dhenderson@gmail.com>

### CODE REVIEW

**Review by bboreham — APPROVED** [review]
lgtm; one point to consider

**Comment by hairyhenderson** [comment]
> one point to consider

Did you miss submitting a comment? I don't see anything 😅

File: tracing/tracing.go [hunk 1]
```diff
@@ -14,9 +14,13 @@ var (
 )
 
 // installJaeger registers Jaeger as the OpenTracing implementation.
-func installJaeger(serviceName string, cfg *jaegercfg.Configuration) (io.Closer, error) {
+func installJaeger(serviceName string, cfg *jaegercfg.Configuration, options ...jaegercfg.Option) (io.Closer, error) {
 	metricsFactory := jaegerprom.New()
-	closer, err := cfg.InitGlobalTracer(serviceName, jaegercfg.Metrics(metricsFactory))
+
+	// put the metricsFactory earlier so provided options can override it
+	opts := append([]jaegercfg.Option{jaegercfg.Metrics(metricsFactory)}, options...)
```
Comment (bboreham) [inline tracing/tracing.go:21]: Well, I don't know what happened, but this is what I was going to suggest:
```suggestion
	opts := append([]jaegercfg.Option{}, jaegercfg.Metrics(metricsFactory), options...)
```

It's perhaps a personal preference: I think it makes it more clear the options form one list.

Comment (hairyhenderson) [reply tracing/tracing.go:21]: @bboreham that doesn't compile 😅

See https://groups.google.com/g/golang-nuts/c/R5RW0oOGSKg for background on why

**Pull request merged**